File: server/src/main/java/com/smartexpense/api/infrastructure/persistence/mapper/UserMapper.java
package com.smartexpense.api.infrastructure.persistence.mapper;

import com.smartexpense.api.domain.model.User;
import com.smartexpense.api.infrastructure.persistence.entity.UserEntity;
import org.springframework.stereotype.Component;

@Component
public class UserMapper {

    public User toDomain(UserEntity entity) {
        if (entity == null) {
            return null;
        }
        return User.builder()
                .id(entity.getId())
                .email(entity.getEmail())
                .passwordHash(entity.getPasswordHash())
                .displayName(entity.getDisplayName())
                .avatarUrl(entity.getAvatarUrl())
                .monthlyBudget(entity.getMonthlyBudget())
                .streakDays(entity.getStreakDays())
                .themeColor(entity.getThemeColor())
                .createdAt(entity.getCreatedAt())
                .updatedAt(entity.getUpdatedAt())
                .build();
    }

    public UserEntity toEntity(User domain) {
        if (domain == null) {
            return null;
        }
        return UserEntity.builder()
                .id(domain.getId())
                .email(domain.getEmail())
                .passwordHash(domain.getPasswordHash())
                .displayName(domain.getDisplayName())
                .avatarUrl(domain.getAvatarUrl())
                .monthlyBudget(domain.getMonthlyBudget())
                .streakDays(domain.getStreakDays())
                .themeColor(domain.getThemeColor())
                .createdAt(domain.getCreatedAt())
                .updatedAt(domain.getUpdatedAt())
                .build();
    }
}

File: server/bin/main/com/smartexpense/api/infrastructure/persistence/mapper/UserMapper.kt
package com.smartexpense.api.infrastructure.persistence.mapper

import com.smartexpense.api.domain.model.User
import com.smartexpense.api.infrastructure.persistence.entity.UserEntity
import org.springframework.stereotype.Component

@Component
class UserMapper {

    fun toDomain(entity: UserEntity) = User(
        id = entity.id,
        email = entity.email,
        passwordHash = entity.passwordHash,
        displayName = entity.displayName,
        avatarUrl = entity.avatarUrl,
        monthlyBudget = entity.monthlyBudget,
        streakDays = entity.streakDays,
        themeColor = entity.themeColor,
        createdAt = entity.createdAt,
        updatedAt = entity.updatedAt
    )

    fun toEntity(domain: User) = UserEntity(
        id = domain.id,
        email = domain.email,
        passwordHash = domain.passwordHash,
        displayName = domain.displayName,
        avatarUrl = domain.avatarUrl,
        monthlyBudget = domain.monthlyBudget,
        streakDays = domain.streakDays,
        themeColor = domain.themeColor,
        createdAt = domain.createdAt,
        updatedAt = domain.updatedAt
    )
}
